docs/proposals/formatted-output.md

```diff
@@ -556,6 +565,117 @@ localhost:5000/kubernetes/kubectl@sha256:bece4f4746a39cb39e38451c70fa5a1e5ea4fa2
                     └── sha256:d2cb66a53e4d77488df1f15701554ebb11ffa1cf6eb90f79afa33d3b172f11d2
 ```
 
+### oras repo ls
+
+When using `oras repo ls` with the `--format` flag, the JSON output contains the following fields:
+
+- `registry`: The name of the registry (e.g., `example.com`).
+- `repositories`: A list of repositories in the registry.
+
+The JSON schema for the output is as follows:
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "JSON Output of Repositories in a Registry",
+  "type": "object",
+  "properties": {
+    "registry": {
+      "type": "string",
+      "description": "The registry name"
+    },
+    "repositories": {
+      "type": "array",
+      "description": "List of repositories in the registry",
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": ["registry", "repositories"],
+  "additionalProperties": false
+}
+```
+
+For example, to list all repositories in a registry and format the output as JSON:
+
+```console
+$ oras repo ls localhost:5000 --format json
+{
+  "registry": "localhost:5000",
+  "repositories": [
+    "dev/foo",
+    "dev/bar",
+    "test/foo"
+  ]
+}
+```
+
+When listing repositories under a specific namespace, the JSON output provides the full repository names, including the namespace prefix. This differs from the default text output, which only shows sub-repository names.
+
+For example, listing repositories under the `dev` namespace with JSON formatting:
+
+```console
+$ oras repo ls localhost:5000/dev --format json
+{
+  "registry": "localhost:5000",
+  "repositories": [
+    "dev/foo",
+    "dev/bar"
+  ]
+}
+```
+
+The default text output for the same command omits the namespace prefix for better readability:
```

Comment:
seems like it would be more consistent if text had the namespace

Reply:
Yes I agree. But the concern is that changing the default output would be breaking. Here is the historical discussion on why the namespace is omitted: https://github.com/oras-project/oras/pull/768#issuecomment-1407609355